src/Compute/Compute/StorageServices/AddAzureVhdCommand.cs

```diff
@@ -228,13 +238,19 @@ public override void ExecuteCmdlet()
 
                         // 3-4: UPLOAD                  
                         WriteVerbose("Preparing for Upload");
-                        PSPageBlobClient managedDisk = new PSPageBlobClient(sasUri);
+                        ComputeTokenCredential tokenCredential = null;
+                        if (this.DataAccessAuthMode == "AzureActiveDirectory")
+                        {
+                            // get token 
+                            tokenCredential = new ComputeTokenCredential(DefaultContext, "https://disk.compute.azure.com/");
```

Comment:
is this audience guaranteed to be same across different cloud?

Reply:
good question